### PR TITLE
[c] Better explanation for ordered dict

### PR DESCRIPTION
**Pull request by catriuspham**
### Related issue

https://github.com/EastAgile/robber.py/issues/48

### Description

Ordered dicts are now handled the same as dicts

### Usage

```python
In [1]: d = {'a': '1', 'b': '2', 'c': '3'}
In [2]: od = OrderedDict({'a': '2', 'b': '2', 'c': '3'})
In [3]: expect(d).to.eq(od1)

Out: 
BadExpectation: 
A = {'a': '1', 'b': '2', 'c': '3'}
B = {'a': '2', 'b': '2', 'c': '3'}
Expected A to equal B
A['a'] = 1 while B['a'] = 2
```

### CODE REVIEW

**Review by hieueastagile — APPROVED** [review]
LGTM, anyway, the piece of code in equal.py might need to be refactored, the casting (cast to list or cast to dict) is repetitive, I think.

**Comment by catriuspham** [comment]
@hieueastagile This should have been based on `feature/unicode-comparison`, sorry if this got you any trouble reviewing the code. Anyway, I gonna make the code in `base.py` look better.

**Comment by catriuspham** [comment]
@hieueastagile I made some update to avoid the repetitiveness

File: robber/matchers/equal.py [hunk 1]
```diff
@@ -5,6 +5,13 @@
 from robber.helper import unicode_to_str
 from robber.matchers.base import Base
 
+try:
+    from collections import OrderedDict
+except ImportError:
+    ordered_dict_available = False
+else:
+    ordered_dict_available = True
```
Comment (hieueastagile) [inline robber/matchers/equal.py:12]: Are you sure that there's sometime ordered_dict is not available. I thought that any python env should come with it?

Comment (catriuspham) [reply robber/matchers/equal.py:12]: https://docs.python.org/2/library/collections.html 
According to this, OrderedDict is a new feature of Python 2.7, but we support Python 2.6, so...

Comment (hieueastagile) [reply robber/matchers/equal.py:12]: I see, so let's merge this first, thanks @catriuspham

**Review by hieueastagile — APPROVED** [review]

**Comment by catriuspham** [comment]
@oyster Please help merge this PR and close this issue: https://github.com/EastAgile/robber.py/issues/48. Thanks.

**Pull request merged**